src/features/walletconnect/services/WalletConnectWallet.ts

```diff
@@ -82,15 +82,10 @@ class WalletConnectWallet {
   }
 
   private getNamespaces(proposal: Web3WalletTypes.SessionProposal, currentChainId: string, safeAddress: string) {
-    // Most dApps require mainnet, but we aren't always on mainnet
-    // As workaround, we pretend include all required and optional chains with the Safe chainId
+    // As workaround, we pretend to support all the required chains plus the current Safe's chain
     const requiredChains = proposal.params.requiredNamespaces[EIP155]?.chains || []
-    const optionalChains = proposal.params.optionalNamespaces[EIP155]?.chains || []
 
-    const supportedChainIds = [currentChainId].concat(
-      requiredChains.map(stripEip155Prefix),
-      optionalChains.map(stripEip155Prefix),
-    )
+    const supportedChainIds = [currentChainId].concat(requiredChains.map(stripEip155Prefix))
```

Comment:
If we don't add the optional chains wouldn't that mean that the moment you switch your network the connection will be closed?

Reply:
I will check

Reply:
You are right. I will readd those :)

Reply:
You can not return a chainId in the namespace without an account. So we would be stuck with the same workaround in the end which caused the bug.
I feel like having to reconnect after switching to another Safe is the lesser devil than not being able to connect on any chain other than mainnet. But We should discuss this.